main/solution/backend/config/infra/cloudformation.yml

```diff
@@ -262,6 +263,121 @@ Resources:
               StringNotEquals:
                 s3:signatureversion: 'AWS4-HMAC-SHA256'
 
+  EgressStoreEncryptionKey:
+    Condition: EnableEgressStore
+    Type: AWS::KMS::Key
+    Properties:
+      Description: >-
+        Master key used to encrypt objects stored in the
+        ${self:custom.settings.egressStoreBucketName} bucket
+      KeyPolicy:
+        Version: '2012-10-17'
+        Id: egress-store-kms-policy
+        Statement:
+          - Sid: Allow root
+            Effect: Allow
+            Principal:
+              AWS:
+                - !Sub 'arn:aws:iam::${AWS::AccountId}:root'
+            Action:
+              - 'kms:*'
+            Resource: '*'
+          - Sid: Allow API access to create object and update policy for new workspaces
+            Effect: Allow
+            Principal:
+              AWS:
+                - !GetAtt [RoleApiHandler, Arn]
+            Action:
+              - kms:GenerateDataKey
+              - kms:DescribeKey
+              - kms:GetKeyPolicy
+              - kms:PutKeyPolicy
+            Resource: '*'
```

Comment:
Is it possible to have a specific kms key here which we will use for egress store?

Reply:
The '*' resource here is allowing different types of workspace resource to use KMS action which means we can not make them as predefined. FYI, we are using similar method in StudyDataEncryptionKey line120.